core/src/main/java/io/questdb/griffin/engine/functions/date/TimestampAddWithTimezoneFunctionFactory.java

```diff
@@ -117,8 +117,8 @@ private static long compute(long timestamp, TimeZoneRules timeZoneRules, int str
             return Numbers.LONG_NULL;
         }
         long offset = timeZoneRules.getOffset(timestamp);
-        long t = periodAddFunction.add(timestamp + offset, stride);
-        return t - timeZoneRules.getOffset(t);
+        long localTimestamp = periodAddFunction.add(timestamp + offset, stride);
+        return localTimestamp - timeZoneRules.getLocalOffset(localTimestamp);
```

Comment:
Is it a breaking change/bug fix?

Reply:
The end effect here is the same as in `to_utc()`, i.e. we convert a local timestamp to UTC, so it's a bugfix. Added more tests around it in a24e67246d892296fc3eea2d21d54b4760503a68